src/euphorie/client/browser/training.py

```diff
@@ -12,43 +19,207 @@ class TrainingSlide(BrowserView):
+            self.context.zodb_path.split("/")
+        )
+
+    @property
+    @memoize
+    def for_download(self):
+        return "for_download" in self.request
 
-class TrainingView(BrowserView):
+    @property
+    def number(self):
+        if self.is_custom:
+            num_elems = self.context.number.split(".")
+            number = u".".join([u"Ω"] + num_elems[1:])
+        else:
+            number = self.context.number
+        if self.item_type == "module":
+            number = u"{}.0".format(number)
+        return number
+
+    @property
+    def slide_title(self):
+        if self.is_custom and self.item_type == "module":
+            return client_utils.get_translated_custom_risks_title(self.request)
+        return self.context.title
+
+    @property
+    def slide_template(self):
+        if self.item_type == "module":
+            return "template-module"
+        if self.description or self.image:
+            return "template-two-column"
+        return "template-default"
+
+    @property
+    def slide_type(self):
+        # Perhaps this will not be needed any more...
+        return self.item_type
+
+    @property
+    def slide_date(self):
+        return date.today().strftime("%Y-%m-%d")
+
+    @property
+    def department(self):
+        # XXX tbd - maybe in webhelpers?
+        return "-"
+
+    @property
+    @memoize
+    def description(self):
+        if self.is_custom:
+            return getattr(self.context, "custom_description", "") or ""
+        return self.zodb_elem.description or ""
+
+    @property
+    def training_notes(self):
+        training_notes = getattr(self.context, "training_notes", "") or ""
+        if self.for_download:
+            training_notes = markdown.markdown(training_notes)
+        return training_notes
+
+    @property
+    @memoize
+    def existing_measures(self):
+        try:
+            existing_measures = loads(self.context.existing_measures)
+            # Backwards compat. We used to save dicts in JSON before we
+            # switched to list of tuples.
+            if isinstance(existing_measures, dict):
+                existing_measures = [(k, v) for (k, v) in existing_measures.items()]
+            existing_measures = [
+                text.strip() for (text, active) in existing_measures if active
+            ]
+        except:
+            existing_measures = []
+        return existing_measures
+
+    @property
+    @memoize
+    def image(self):
+        if self.is_custom:
+            if getattr(self.context, "image_data", None):
+                _view = self.context.__of__(
+                    self.webhelpers.traversed_session.aq_parent["custom-risks"]
+                ).restrictedTraverse("image-display")
+                _view.field_name = "image_large"
+                return _view.get_or_create_image_scaled()
+            return None
+        image = self.zodb_elem.image and self.zodb_elem.image.data or None
```

Comment:
It seems to me there is no point in getting the image if `self.for_download` is False.